subsys/bluetooth/host/gatt.c

```diff
@@ -640,11 +647,13 @@ static ssize_t db_hash_read(struct bt_conn *conn,
 			    const struct bt_gatt_attr *attr,
 			    void *buf, u16_t len, u16_t offset)
 {
+	int err;
+
 	/* Check if db_hash is already pending in which case it shall be
-	 * generated immediately instead of waiting the work to complete.
+	 * generated immediately instead of waiting for the work to complete.
 	 */
-	if (k_delayed_work_remaining_get(&db_hash_work)) {
-		k_delayed_work_cancel(&db_hash_work);
+	err = k_delayed_work_cancel(&db_hash_work);
```

Comment:
Interesting, I recall k_delayed_work_cancel was not causing any errors if it was already completed but it looks like that has been changed since then.